src/ty.md

```diff
@@ -2,94 +2,96 @@
+information.
 
-**Order** HIR is built directly from the AST, so it happens before any `ty::Ty` is produced. After
-HIR is built, some basic type inference and type checking is done. During the type inference, we
-figure out what the `ty::Ty` of everything is and we also check if the type of something is
-ambiguous. The `ty::Ty` then, is used for type checking while making sure everything has the
-expected type. The [`astconv` module][astconv] is where the code responsible for converting a
-`rustc_hir::Ty` into a `ty::Ty` is located. This occurs during the type-checking phase,
-but also in other parts of the compiler that want to ask questions like "what argument types does
-this function expect?"
+The [`astconv`][astconv] module handles the actual conversion of a
+[`rustc_hir::Ty`][hir_ty] into a [`ty::Ty`] during the type-checking phase, or
+in other parts of the compiler which need specific type-based information such
+as (but not limited to): "what argument types does this function expect?"
 
-[astconv]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_typeck/astconv/index.html
+**Semantics of `Ty`** 
 
-**How semantics drive the two instances of `Ty`** You can think of HIR as the perspective
-of the type information that assumes the least. We assume two things are distinct until they are
-proven to be the same thing. In other words, we know less about them, so we should assume less about
-them.
+[HIR](hir.md) assumes the least about type information. Since
+[`rustc_hir::Ty`](hir_ty) knows very little about type distinction it assumes
+much less about types than [`ty::Ty`] which is able to prove indistinction and
+make specific assumptions.
 
-They are syntactically two strings: `"u32"` at line N column 20 and `"u32"` at line N column 35. We
-don’t know that they are the same yet. So, in the HIR we treat them as if they are different. Later,
-we determine that they semantically are the same type and that’s the `ty::Ty` we use.
+Consider another example: `fn foo<T>(x: T) -> u32`. Someone then invokes
+`foo::<u32>(0)`. This means that `T` and [`u32`][u32] resolve to the same
+[`ty::Ty`] but each have a distinct record in [`rustc_hir::Ty`][hir_ty].
 
-Consider another example: `fn foo<T>(x: T) -> u32`. Suppose that someone invokes `foo::<u32>(0)`.
-This means that `T` and `u32` (in this invocation) actually turns out to be the same type, so we
-would eventually end up with the same `ty::Ty` in the end, but we have distinct `rustc_hir::Ty`.
-(This is a bit over-simplified, though, since during type checking, we would check the function
-generically and would still have a `T` distinct from `u32`. Later, when doing code generation,
-we would always be handling "monomorphized" (fully substituted) versions of each function,
-and hence we would know what `T` represents (and specifically that it is `u32`).)
+In the actually implemented type checking the function is actually still checked
```

Comment:
nit: "In the actually implemented type checking" is a bit of an awkward way to phrase it. Maybe just remove it?